sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThroughputControlGroup.java

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.util.Beta;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Group configuration which will be used in Throughput control.
+ */
+@Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public class ThroughputControlGroup {
+
+    private final static boolean DEFAULT_USE_BY_DEFAULT = false;
+    private final static ThroughputControlMode DEFAULT_CONTROL_MODE = ThroughputControlMode.LOCAL;
+
+    private final String groupName;
+    private final String id;
+    private final CosmosAsyncContainer targetContainer;
+
+    private ThroughputControlMode controlMode;
+    private Integer targetThroughput;
+    private Double targetThroughputThreshold;
+    private boolean useByDefault;
+
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ThroughputControlGroup(String groupName, CosmosAsyncContainer targetContainer, int targetThroughput) {
+        this(groupName, targetContainer, targetThroughput, null);
+    }
+
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ThroughputControlGroup(String groupName, CosmosAsyncContainer targetContainer, double targetThroughputThreshold) {
```

Comment:
I'm still not sure why you need a CosmosAsyncContainer here... the enableThroughputControl is at the CosmosAsynClient level anyway so it has all necessary the context you need. A database and container name as a string should be enough for now.
Is the intention here to persist the current RU into a different container for better synchronization across different Cosmos clients?

Reply:
As discussed in the team meeting, will using the following proposed way:
ThroughputControlGroup group = syncContainer.enableLocalThroughputControlGroup(groupName, targetThroughput);
        ThroughputControlGroup group = syncContainer.enableDistributedThroughputControlGroup(groupName, targetThroughput, distributedConfig);